lib/utils/manager/upload_manager/upload_manager.dart

```diff
@@ -176,14 +214,18 @@ class UploadManager {
     Map<MatrixFile, MatrixImageFile?>? thumbnails,
     String? caption,
   }) async {
-    for (final MatrixFile matrixFile in files) {
+    for (final matrixFile in files.asMap().entries) {
```

Comment:
why need to convert to map ?

Reply:
final index = matrixFile.key;
final matrixFile = matrixFile.value;

Reply:
Fixed